### PR TITLE
Feat/Add MSTL

### PR DESCRIPTION
**Pull request by alexcolpitts96**
Adds MSTL as requested in #1249. I accidentally deleted my branch meant for #1404.

### CODE REVIEW

File: darts/utils/statistics.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def _bartlett_formula(r: np.ndarray, m: int, length: int) -> float:
 
 def extract_trend_and_seasonality(
     ts: TimeSeries,
-    freq: int = None,
+    freq: Union[int, list] = None,
```
Comment (madtoinou) [inline darts/utils/statistics.py:126]:
```suggestion
    freq: Union[int, List] = None,
```

**Review by madtoinou — CHANGES_REQUESTED** [review]
Thank you for being on top of the statsmodels releases.

Could you please also update the core.txt requirements with `statsmodels>=0.14.0` to make sure `MSTL` is available?

File: darts/utils/statistics.py [hunk 2]
```diff
@@ -189,6 +192,19 @@ def extract_trend_and_seasonality(
             **kwargs,
         ).fit()
 
+    elif method == "MSTL":
+        raise_if_not(
+            model in [SeasonalityMode.ADDITIVE, ModelMode.ADDITIVE],
+            f"Only ADDITIVE model is compatible with the MSTL method. Current model is {model}.",
+            logger,
+        )
+
```
Comment (madtoinou) [inline darts/utils/statistics.py:201]: Also check that freq is a `List` for only if `method == "MSTL"` before the if/else.

Comment (alexcolpitts96) [reply darts/utils/statistics.py:206]: Implemented and updated testing.

Comment (alexcolpitts96) [reply darts/utils/statistics.py:206]: Updated and added tests.

File: darts/utils/statistics.py [hunk 3]
```diff
@@ -170,6 +173,11 @@ def extract_trend_and_seasonality(
         "The model must be either MULTIPLICATIVE or ADDITIVE.",
     )
 
+    raise_if_not(
+        (method == "MSTL") & (isinstance(freq, list)) or (method != "MSTL"),
+        f"MSTL method requires a list of frequencies. Got a single value instead: {freq}.",
+    )
```
Comment (madtoinou) [inline darts/utils/statistics.py:179]: Sorry, my previous comment was not specific enough, what I meant was if the method is not "MSTL", we show an elegant message saying that the method (`f"... {method} ..."`) only support integer.

File: darts/utils/statistics.py [hunk 4]
```diff
@@ -165,11 +168,18 @@ def extract_trend_and_seasonality(
...
     )
+
     raise_if_not(
         model is not SeasonalityMode.NONE,
         "The model must be either MULTIPLICATIVE or ADDITIVE.",
     )
 
+    raise_if_not(
+        isinstance(freq, int) or method == "MSTL",
+        f"Method {method} is not compatible with freq {freq}. Only MSTL can use multiple seasonalities.",
```
Comment (madtoinou) [inline darts/utils/statistics.py:179]:
```suggestion
    raise_if(
        isinstance(freq, Sequence) and method != "MSTL",
        f"{method} decomposition cannot be performed with more than one seasonality, received {freq}.",
```

**Review by madtoinou — COMMENTED** [review]
Almost ready for merge!

The tests are failing due to the sanity check on `freq`, I suggested some changes.

File: darts/tests/utils/test_statistics.py [hunk 5]
```diff
@@ -136,12 +136,43 @@ def test_extract(self):
...
+                self.ts, freq=[1, 4, 6], method="STL", model=ModelMode.ADDITIVE
+            )
+
+        # check if error is raised when using multiplicative model
         with self.assertRaises(ValueError):
             calc_trend, _ = extract_trend_and_seasonality(
                 self.ts, freq=6, method="STL", model=ModelMode.MULTIPLICATIVE
             )
 
+        # make sure non MSTL methods fail with multiplicative model
```
Comment (madtoinou) [inline darts/tests/utils/test_statistics.py:170]: the comment does not match the test, should be removed (covered by the comment above)

**Review by madtoinou — APPROVED** [review]
One very little change in the test and it should be ready for merge 🚀 

Thanks for the work!

**Review by dennisbader — APPROVED** [review]
Thanks @alexcolpitts96 for another neat contribution 🚀 
Can you add it to the CHANGELOG.md? After that we can merge :)

**Comment by alexcolpitts96** [comment]
Done!

**Pull request merged**